test/telemetry/src/main/java/org/opensearch/test/telemetry/tracing/MockSpanData.java

```diff
@@ -0,0 +1,175 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.test.telemetry.tracing;
+
+import java.util.Arrays;
+
+/**
+ * MockSpanData model for storing Telemetry information for testing.
+ */
+public class MockSpanData {
```

Comment:
Why do you need this class? The spans are fully extensible by means of attributes, the OTels SpanData should be fully sufficient

Reply:
OTels SpanData is available in OTel sdk which is present in plugin only, test framework does not have its dependency. Wanted to write common validator framework (which is in test framework) for MockTelemetryPlugin and OtelTelemetryPlugin, so introduced a new class which validator understands. Both the plugins are writing/converting data to this MockSpanData object.